internal/server/api.yaml

```diff
@@ -239,21 +239,50 @@ components:
       required:
         - status
         - type
+        - options
```

Comment:
Hm I know we marked this as required in composer as well, but really it'll only contain meaningful data if the status property === 'success', just a thought.

Reply:
Yes, your comment makes sense. I probably didn't hit this issue, because the upload always passed in CI so far... I'll change it to not be required (and test it with e.g. non existent GCP account for sharing). Another thought I just had while writing documentation for image builder API is that it seems that we don't return any errors, just status. This may not be ideal for the user because they may not be able to figure out the reason for the failure...

Reply:
So this is not really a problem, because the `upload_status` is not part of the response at all in case the upload failed. See https://osbuildci.cloud.paas.psi.redhat.com/blue/organizations/jenkins/osbuild%2Fimage-builder/detail/PR-156/9/pipeline#step-36-log-788

Reply:
Alright keep it as required then, I think it that makes sense